modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/fetch.es.js

```diff
@@ -22,22 +22,28 @@
  */
 
 export default function defaultFetch(resource, init = {}) {
+	const pathContext = themeDisplay.getPathContext();
+
 	if (!resource) {
-		resource = '/o/';
+		resource = `${pathContext}/o/`;
 	}
 
 	let resourceLocation = resource.url ? resource.url : resource.toString();
 
 	if (resourceLocation.startsWith('/')) {
-		resourceLocation = window.location.origin + resourceLocation;
+		resourceLocation =
+			window.location.origin + pathContext + resourceLocation;
 	}
 
 	const resourceURL = new URL(resourceLocation);
 
 	const headers = new Headers({});
 	const config = {};
 
-	if (resourceURL.origin === window.location.origin) {
+	if (
+		resourceURL.origin + pathContext ===
+		window.location.origin + pathContext
```

Comment:
nit: Isn't adding the same `pathContext` here irrelevant? It'd be like checking `someVar + '123' === anotherVar + '123'` instead of `someVar === anotherVar`

Reply:
You may need to make sure that a site can't "fake" the path though and get access to the authToken

So we need to be sure that if you have two sites on the same origin, that site A without the pathContext can't get access to the authToken for the site with the pathContext

Reply:
Same origin policy is scoped to the concept of [origin](https://developer.mozilla.org/en-US/docs/Glossary/Origin) which involves protocol, server and port. Anything after it is irrelevant:

```
> new URL("http://a.com:80/a/b/c#hash?x=1&y=2").origin
<- 'http://a.com'
```
So, testing `.origin` should be fine to enforce the policy

Reply:
> nit: Isn't adding the same pathContext here irrelevant? It'd be like checking someVar + '123' === anotherVar + '123' instead of someVar === anotherVar

Yes, it's irrelevant. Changing :)

I just considered to fix this issue appending pathContext to window.location.origin but in comparison it doesn't makes sense. Also it can be considered a tautology :p
![image](https://user-images.githubusercontent.com/7663513/166507823-79048d36-9742-4aa2-a06b-4c455792dcc2.png)